### PR TITLE
Display save directory in statusbar of Eng Diff UI

### LINKED ISSUE #29063: Add save directory to display in status bar of engineering diffraction UI
The save directory is accessible via the settings menu, however many users will not try to modify or view the settings. It would be helpful to add the save directory to the status bar so it is always visible.

Report to: Saurabh (ENGIN-X)

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
Display save directory in the status bar of the Engineering diffraction UI (bottom of window)

**To test:**

(1) Interfaces > Engineering Diffraction
(2) Click the cog button at the bottom left of the window to open settings
(3) Check the save directory in the settings matches that in the status bar (bottom of UI window)
(4) Browse to a different save directory 
(5) Apply or save
(6) Check the save directory in the status bar has updated correctly

The save directory box has a max length, if the length of the directory string exceeds this you should still be able to view the full path by hovering over the mouse over it to display the tool tip.

Fixes #29063 

*This does not require release notes* because **very minor change to info displayed**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by cailafinn — DISMISSED** [review]
Just noticed the one thing in the code, otherwise works well.

Could you also add this new functionality to the eng diff docs?

File: scripts/Engineering/gui/engineering_diffraction/settings/settings_presenter.py [hunk 1]
```diff
@@ -43,6 +45,7 @@ def save_and_close_dialog(self):
 
     def save_new_settings(self):
         self._collect_new_settings_from_view()
+        self.savedir_notifier.notify_subscribers(self.settings["save_location"])
```
[the remark this answers is not in the document]
Comment (RichardWaiteSTFC) [reply scripts/Engineering/gui/engineering_diffraction/settings/settings_presenter.py]: Good spot, I have no put notify_subscribers in another function post validation of path

File: scripts/Engineering/gui/engineering_diffraction/engineering_diffraction.py [hunk 2]
```diff
@@ -45,23 +46,27 @@ def __init__(self, parent=None):
...
         # Setup Elements
         self.setup_settings()
         self.setup_calibration()
         self.setup_focus()
         self.setup_fitting()
 
+        # Setup status bar
+        self.status_label = QtWidgets.QLabel()
+        self.savedir_label = QtWidgets.QLabel()
+        self.savedir_label.setMaximumWidth(300)
```
Comment (clayton-tom) [inline scripts/Engineering/gui/engineering_diffraction/engineering_diffraction.py:64]: Definitely a nitpick, but would it be preferable to declare a const elsewhere for the max width pixels and avoid this magic number?

**Review by clayton-tom — CHANGES_REQUESTED** [review]
Looks good! Found one minor change to potentially help someone looking at this in the future

File: scripts/Engineering/gui/engineering_diffraction/engineering_diffraction.py (hunk 2 above)
Comment (RichardWaiteSTFC) [reply scripts/Engineering/gui/engineering_diffraction/engineering_diffraction.py]: Done!

**Review by clayton-tom — APPROVED** [review]

**Comment by clayton-tom** [comment]
Works fine and all nice and clean, good job!

**Pull request merged**